pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -336,18 +336,21 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 
 	// Explicitly assign mac address to cbr0. If bridge mac address is not explicitly set will adopt the lowest MAC address of the attached veths.
 	// TODO: Remove this once upstream cni bridge plugin handles this
-	link, err := netlink.LinkByName(BridgeName)
-	if err != nil {
-		return fmt.Errorf("failed to lookup %q: %v", BridgeName, err)
-	}
-	macAddr, err := generateHardwareAddr(plugin.gateway)
-	if err != nil {
-		return err
-	}
-	glog.V(3).Infof("Configure %q mac address to %v", BridgeName, macAddr)
-	err = netlink.LinkSetHardwareAddr(link, macAddr)
-	if err != nil {
-		return fmt.Errorf("Failed to configure %q mac address to %q: %v", BridgeName, macAddr, err)
+	var macAddr net.HardwareAddr
+	if plugin.gateway.To4() != nil {
```

Comment:
I can generate a random mac, if that is preferred, but doesn't the link get a random mac assigned when it is created? Here is the cbr0 for IPv6 only using local_cluster_up
16052: cbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 9a:5d:36:a0:3a:e2 brd ff:ff:ff:ff:ff:ff
    inet6 2001:101::1/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::5c2a:e0ff:fed5:75be/64 scope link 
       valid_lft forever preferred_lft forever